PFCalEE/userlib/src/HGCSSGeometryConversion.cc

```diff
@@ -119,24 +128,82 @@ void HGCSSGeometryConversion::initialiseSquareMap(TH2Poly *map, const double xym
   
 }
 
-void HGCSSGeometryConversion::initialiseHoneyComb(const double xymin, const double side){
-  initialiseHoneyComb(hexagonMap(),xymin,side,true);
+void HGCSSGeometryConversion::initialiseHoneyComb(const double width, const double side){
+  initialiseHoneyComb(hexagonMap(),width,side,true);
   fillXY(hexagonMap(),hexaGeom);
 }
 
-void HGCSSGeometryConversion::initialiseHoneyComb(TH2Poly *map, const double xymin, const double side, bool print){
+void HGCSSGeometryConversion::initialiseHoneyComb(TH2Poly *map, const double width, const double side, bool print){
   //xstart,ystart,side length,
-  double d=sqrt(3.)*side;
-  unsigned nx=static_cast<unsigned>(xymin*2./d);
-  unsigned ny=static_cast<unsigned>(xymin*4./(3.*side));
+
+  // Center a cell at (x,y)=(0,0) and ensure coverage up to/past width/2 in all 4 directions,
+  // assuming each cell is lying on a side.
+
+  unsigned ncellwide=11;
```

Comment:
The thing is that this method is called by everyone, not just for the 
TB2016 setup, so if we want a larger setup, we don't want that "11" to 
be hardcoded.



On 04/05/2016 08:46 PM, pdudero wrote:
> In PFCalEE/userlib/src/HGCSSGeometryConversion.cc
> <https://github.com/pfs/PFCal/pull/17#discussion_r58594071>:
>
>>    //xstart,ystart,side length,
>> -  double d=sqrt(3.)*side;
>> -  unsigned nx=static_cast<unsigned>(xymin*2./d);
>> -  unsigned ny=static_cast<unsigned>(xymin*4./(3.*side));
>> +
>> +  // Center a cell at (x,y)=(0,0) and ensure coverage up to/past width/2 in all 4 directions,
>> +  // assuming each cell is lying on a side.
>> +
>> +  unsigned ncellwide=11;
>
> Could be. In fact "11" also shows up in the constructor at the top.
> However, I don't know what the hardware plans are, whether the sensor
> size will change after this.
> … <#>
> On 4/5/2016 2:47 AM, amagnan wrote: In
> PFCalEE/userlib/src/HGCSSGeometryConversion.cc <#17 (comment)
> <https://github.com/pfs/PFCal/pull/17#discussion_r58498966>>: >
> //xstart,ystart,side length, > - double d=sqrt(3.)*side; > - unsigned
> nx=static_cast<unsigned>(xymin*2./d); > - unsigned
> ny=static_cast<unsigned>(xymin*4./(3.*side)); > + > + // Center a cell
> at (x,y)=(0,0) and ensure coverage up to/past width/2 in all 4
> directions, > + // assuming each cell is lying on a side. > + > +
> unsigned ncellwide=11; I think ncellwide should be given as argument
> (instead of "width" which is not used...) and to avoid hardcoding the
> "11" ... — You are receiving this because you authored the thread. Reply
> to this email directly or view it on GitHub
> <https://github.com/pfs/PFCal/pull/17/files/10d68cea58df7a3601955203a93d7cc949b732a4#r58498966>
>
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/pfs/PFCal/pull/17/files/10d68cea58df7a3601955203a93d7cc949b732a4#r58594071>
>

-- 

####################################################
Dr. Anne-Marie Magnan
CMS

----------------------   ---------------------------
42-3-037, CERN            Imperial College, HEP Dept.
CH-1210 Geneva            The Blackett Laboratory
                           Prince Consort Road
                           London SW7 2AZ
(+41)(0)22 76 6 25 46
http://www.cern.ch        http://www.imperial.ac.uk

####################################################
â à è é ê ë æ î ï ô ö ù û ü ç
####################################################